prefect_kubernetes/worker.py

```diff
@@ -307,16 +307,37 @@ def prepare_for_flow_run(
         super().prepare_for_flow_run(flow_run, deployment, flow)
         # Update configuration env and job manifest env
         self._update_prefect_api_url_if_local_server()
-        self.job_manifest["spec"]["template"]["spec"]["containers"][0]["env"] = [
-            {"name": k, "value": v} for k, v in self.env.items()
-        ]
+        self._populate_env_in_manifest()
         # Update labels in job manifest
         self._slugify_labels()
         # Add defaults to job manifest if necessary
         self._populate_image_if_not_present()
         self._populate_command_if_not_present()
         self._populate_generate_name_if_not_present()
 
+    def _populate_env_in_manifest(self):
+        transformed_env = [{"name": k, "value": v} for k, v in self.env.items()]
```

Comment:
Would it be possible to add a docstring here about how it parses potentially hardcoded environment variables to the base job template to enable the use of Kubernetes secrets?